### ISSUE TITLE
Explain Resnet

### ISSUE THREAD

**Issue by rrajaram2019** [Sachini/ronin#15]
Could you please explain what these lines in ronin_resnet.py are doing? I don't have much experience with torch, but I am wondering where feat, targ, pred, preds_all and targets_all are being pulled from?

def run_test(network, data_loader, device, eval_mode=True):
    targets_all = []
    preds_all = []
    if eval_mode:
        network.eval()
    for bid, (feat, targ, _, _) in enumerate(data_loader):
        pred = network(feat.to(device)).cpu().detach().numpy()
        targets_all.append(targ.detach().numpy())
        preds_all.append(pred)
    targets_all = np.concatenate(targets_all, axis=0)
    preds_all = np.concatenate(preds_all, axis=0)
    return targets_all, preds_all

**Comment by orekides** [comment]
bascially, data_loader is a pytorch object to provide  an iterable over the given dataset.
check the pytorch doc here: https://pytorch.org/docs/stable/data.html?highlight=dataloader#torch.utils.data.DataLoader
feat,targ is a batch of X，Y respectively. the bid here stands for batch index.
so the for-loop is a common process of loop over  batch of data

**Comment by rrajaram2019** [comment]
I'm sorry, I still don't quite understand. So where is the dataset being passed into run_test? 

Network I assume holds the checkpoint/pre-trained model. 

What does "enumerate(data_loader)" do? I get that feat, targ and _,_ are "batches" but what do you mean batches of X,Y?

After feat and targ are extracted, what does this line do: pred = network(feat.to(device)).cpu().detach().numpy()

**Comment by orekides** [comment]
data_loader is initialized with the dataset to iterate through.
for the other questions , maybe you should learn some basics of pytorch/python.
